### PR TITLE
Generate minified .min.js and unminified .js files for GB js entry points when building

### PR DESCRIPTION
**Pull request by fullofcaffeine**
## Description

Attempts to solve https://github.com/WordPress/gutenberg/issues/23960. 

Follow-up to: https://github.com/WordPress/gutenberg/pull/31729.

- [x] Generate an entry for the minified file and another for the unminified version for each entry-point;
- [x] Configure the `TerserPlugin` to only apply to files that match `.min.js`;
- [x] Enqueue `min.js` files by default;
- [x] Add support for WordPress' `SCRIPT_DEBUG` constant. When it's `true`, the unminified version of the script should be loaded, instead.

## How has this been tested?

- [x] Run `npm install` and `npm run build` and verify that for each of the generated packages, there's a `<name>.js` that is NOT minified and `<name>.min.js` that is minified (if it's a block, check that the same applies for the `frontend.[min].js` script) for each entry-point in `build/`.
- [x] Verify that the Gutenberg plugin is working correctly and that these changes do not break it in some unexpected way. The minified files should load by default.
- [x] Verify that `index.min.js` loads when the `SCRIPT_DEBUG` PHP constant is not set or set and false, or `index.js` loads when `SCRIPT_DEBUG` is set and true (the const is set to `true` by default in `wp-config.php`).

You can also verify / run the unit test. From the Gutenberg root directory, run: `npm run test-unit packages/readable-js-assets-webpack-plugin/test/build.js`. It should pass 🟢.

## Types of changes

- New WebPack plugin (non-breaking change which adds functionality to the build process);
- Change in how the asset is loaded by the PHP WP assets loader. 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: webpack.config.js
Comment (fullofcaffeine) [inline webpack.config.js]: I'm wondering if we really need to generate a `frontend.min.js`? 🤔

**Comment by fullofcaffeine** [comment]
The downside of this approach is that the build task runs webpack twice (through wp-scripts): one time to generate the minified min.js script (and its asset PHP file) and another pass to generate the unminfied version.

File: lib/client-assets.php
Comment (fullofcaffeine) [inline lib/client-assets.php]: I'm not really sure if we should change the filename to `index.min.js` here 🤔 (since this function seems related to translations). Could someone shed some light? Is it okay to leave it as is?

**Review by WunderBart — APPROVED** [review]
This tests well for me! Thanks for working on this 🥳

> Run npm run build and verify that for each of the generated packages, there's a <name>/index.js that is NOT minified and <name>/index.min.js that is minified (if it's a block, check that the same applies for the frontend.[min].js script).

✅ Verified: `index.min.js` files are minified & `index.js` aren't.

> Verify that the Gutenberg plugin is working correctly and that these changes do not break it in some unexpected way. The minified files should load by default.

✅ Verified: I've built the plugin and uploaded it to my site. Minified files load by default and the plugin seems to be working as usual.

> Verify that index.min.js loads when the SCRIPT_DEBUG PHP constant is not set or set and false, or index.js loads when SCRIPT_DEBUG is set and true (the const is set to true by default in wp-config.php).

✅ Verified: `SCRIPT_DEBUG` constant determines correctly which files should be requested.

---

@gziolo @youknowriad how does it look to you? Is there anything that you'd consider a blocker here?

**Comment by youknowriad** [comment]
Do we know why the original attempt from @sirreal has been reverted?

How can we check that i18n string discovery still works properly with this PR? (I think GlotPress uses some `wp-cli` command to extract these)

**Comment by sirreal** [comment]
The original attempt renamed the single generated JavaScript file extensions from `.js` to `.min.js`. [Producing a single `.min.js` file (without an accompanying `.js` file) was problematic because:](https://github.com/WordPress/gutenberg/issues/23960#issuecomment-666256309)

> The translations were not parsed because the plugin repository ignores `.min.js` when extracting strings for translation.

[Here's a link to the relevant Core Slack conversation.](https://wordpress.slack.com/archives/C02QB2JS7/p1595933189489900)

File: webpack.config.js
Comment (jsnajdr) [inline webpack.config.js]: Does this mean that we are setting up the minimizer even if `MINIMIZE` is false, and just setting it up to ignore the input files?

You could simply set the `optimization.minimize = MINIMIZE` boolean option.

**Review by jsnajdr — COMMENTED** [review]
The two webpack compilations generate two `index.asset.php` and `index.min.asset.php` files that are identical. We need only one. Does the filename need the `.min.` infix?

We should disable the `DependencyExtractionWebpackPlugin` for the non-minified compilation. And if we don't want to include `.min.` in the PHP filename, patch the part of the plugin that does:
```js
assetFilename = buildFilename.replace( /\.js$/, '.asset.php' );
```
and replace the entire `.min.js` suffix (with a `/(\.min)?\.js$/` regexp).

If we want to avoid the double compilation and output both `.js` and `.min.js` assets at the same time, it should be possible with a simple webpack plugin. One that hooks to a place right before the Terser minification, outputs a `.js` asset, and then lets Terser minify the source into a `.min.js` asset. Let me know if you want me to push a commit with that 🙂

**Comment by fullofcaffeine** [comment]
> The two webpack compilations generate two index.asset.php and index.min.asset.php files that are identical. We need only one. Does the filename need the .min. infix?

Oh, good catch!

> We should disable the DependencyExtractionWebpackPlugin for the non-minified compilation. And if we don't want to include .min. in the PHP filename, patch the part of the plugin that does:

Thanks for pointing this out, I wasn't aware of this part.


> If we want to avoid the double compilation and output both .js and .min.js assets at the same time, it should be possible with a simple webpack plugin. One that hooks to a place right before the Terser minification, outputs a .js asset, and then lets Terser minify the source into a .min.js asset. Let me know if you want me to push a commit with that slightly_smiling_face

@jsnajdr That would be great! ❤️  🙇🏻

File: webpack.config.js
Comment (fullofcaffeine) [reply webpack.config.js]: > Does this mean that we are setting up the minimizer even if MINIMIZE is false, and just setting it up to ignore the input files?

Yes, it will run but will ignore due to the match.

> You could simply set the optimization.minimize = MINIMIZE boolean option.

TIL, Thanks!

**Comment by fullofcaffeine** [comment]
> > If we want to avoid the double compilation and output both .js and .min.js assets at the same time, it should be possible with a simple webpack plugin. One that hooks to a place right before the Terser minification, outputs a .js asset, and then lets Terser minify the source into a .min.js asset. Let me know if you want me to push a commit with that slightly_smiling_face

> @jsnajdr That would be great! heart 🙇🏻

Actually, I think it'd be fun to poke with WebPack plugins. I'll take it for a spin today and see if I can manage to get it working.

**Comment by fullofcaffeine** [comment]
> Actually, I think it'd be fun to poke with WebPack plugins. I'll take it for a spin today and see if I can manage to get it working.

@jsnajdr I gave it a try but didn't manage to implement what you suggested, yet :(. I pushed what I did so far in the following commit: https://github.com/WordPress/gutenberg/pull/31732/commits/7049cfacfbc8a56e6eeb9b70f57b73d071077583.

After failing to tap into a hook that would allow me to generate the unminized files *before* the minimizer ran, I tried a more brute-force approach in the [plugin I created](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4R29-R50), but it doesn't work. Well, it **does** generate the `<name>.js`, so the output looks good if you don't see the contents of the corresponding `<name>.min.js`, which are not minimized. I realized the `apply` method [here](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4R46) only hooks the Terser plugin in WebPack's lifecycle, so calling it like this doesn't minimize the files synchronously as I was expecting.

If I could get hold of the unminized  source for each asset at [this point](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4R37-R38), then I could leave Terser where it belongs in the `webpack.min.js`. The idea would be to:

1. Pass the Terser plugin in the `optimizations.minifier` attribute in the main WebPack config object, as it should be passed as per the WebPack API;
2. In our custom plugin, we `tap` into the `compiler.emit` hook (like I'm doing the plugin [here](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4R33)), and generate the corresponding `<name>.js` files for each `<name>.min.js` (like I'm doing [here](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4R34-R40));
3. Each new `<name>.js` entry, we should assign the unminized source code. **And here's where I got stuck** (and that's when I tried the brute-force approach of [manually calling](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4R46) the Terser plugin, which didn't work). I don't know how to get the unminized source code from [here](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4R37-R38); also considering that at this point if the minimizer is active, we'll already get the minimized source for each entry asset, so the the original source is not part of the assets map anymore.

Or maybe I just didn't understand how to tap into the emit hook but *before* the Terser runs. Even if I do, though, I'd have to somehow make sure the asset entries I add don't get mangled later.

I'll continue tomorrow, but if you know how to solve this from the top of your head, feel free to jump in. If you don't have the time, I could use some tips on how to proceed. Thanks in advance!

**Comment by jsnajdr** [comment]
@fullofcaffeine I'm playing with something like this:
```js
class AddReadableJsAssetsWebpackPlugin {
	extractUnminifiedFiles( compilation ) {
		const files = compilation.chunks.flatMap( ( chunk ) => chunk.files );
		compilation.unminifiedAssets = files.map( ( file ) => {
			const asset = compilation.assets[ file ];
			const unminifiedFile = file.replace( /\.min\.js$/, '.js' );
			return [ unminifiedFile, asset.source() ];
		} );
	}
	async writeUnminifiedFiles( compilation ) {
		for ( const [ file, source ] of compilation.unminifiedAssets ) {
			await fs.promises.writeFile( file, source );
		}
	}
	apply( compiler ) {
		compiler.hooks.compilation.tap(
			this.constructor.name,
			( compilation ) => {
				compilation.hooks.additionalAssets.tap(
					this.constructor.name,
					() => this.extractUnminifiedFiles( compilation )
				);
			}
		);
		compiler.hooks.afterEmit.tapPromise(
			this.constructor.name,
			( compilation ) => this.writeUnminifiedFiles( compilation )
		);
	}
}
```
Don't modify anything about `TerserPlugin` at all. Save the asset source files before Terser runs, in the `additionalAssets` hook (runs right before `optimizeAssets`). Write them after all production assets are written, to already-created directories like `build/a11y`.

That's all I managed to do today. You can continue from this point, I'm coming back tomorrow again.

**Comment by fullofcaffeine** [comment]
> @fullofcaffeine I'm playing with something like this:
> 
> ```js
> class AddReadableJsAssetsWebpackPlugin {
> 	extractUnminifiedFiles( compilation ) {
> 		const files = compilation.chunks.flatMap( ( chunk ) => chunk.files );
> 		compilation.unminifiedAssets = files.map( ( file ) => {
> 			const asset = compilation.assets[ file ];
> 			const unminifiedFile = file.replace( /\.min\.js$/, '.js' );
> 			return [ unminifiedFile, asset.source() ];
> 		} );
> 	}
> 	async writeUnminifiedFiles( compilation ) {
> 		for ( const [ file, source ] of compilation.unminifiedAssets ) {
> 			await fs.promises.writeFile( file, source );
> 		}
> 	}
> 	apply( compiler ) {
> 		compiler.hooks.compilation.tap(
> 			this.constructor.name,
> 			( compilation ) => {
> 				compilation.hooks.additionalAssets.tap(
> 					this.constructor.name,
> 					() => this.extractUnminifiedFiles( compilation )
> 				);
> 			}
> 		);
> 		compiler.hooks.afterEmit.tapPromise(
> 			this.constructor.name,
> 			( compilation ) => this.writeUnminifiedFiles( compilation )
> 		);
> 	}
> }
> ```
> 
> Don't modify anything about `TerserPlugin` at all. Save the asset source files before Terser runs, in the `additionalAssets` hook (runs right before `optimizeAssets`). Write them after all production assets are written, to already-created directories like `build/a11y`.
> 
> That's all I managed to do today. You can continue from this point, I'm coming back tomorrow again.

@jsnajdr Oh now I understand what you had in mind! 

To clarify: My idea was to avoid manually getting the file contents and writing to the filesystem myself. I thought it'd be simpler to just hook into the lifecycle to add some unminimized assets to the list before terser runs (or after), and then let the already setup WebPack pipeline run after that, thus spitting all the files for us. As you can see, I did manage to accomplish part of it, I just couldn't find a way to exclude files from being minimized or get unminimized files after the minimizer ran :(

Anyway, your approach is sound and I'll try it out today! Thanks a lot for taking the time to work on it!

**Comment by fullofcaffeine** [comment]
@jsnajdr Your snippet worked pretty much verbatim! _(run `npm run build` to verify)_ 🎉 

There are two things I'd like feedback on:
1. Where will this module live? 
2. What do you think about its name?

For 1), I took the freedom to move it to an inner package under `packages/` in the monorepo (and also added unit tests), [see this commit](https://github.com/WordPress/gutenberg/pull/31732/commits/93339727e5e9757bf2a7f3230789303d171280d7). I'm not 100% sure if this would be the best way to go here, but by looking at [other](https://github.com/WordPress/gutenberg/tree/trunk/packages/dependency-extraction-webpack-plugin) custom plugins we developed for WebPack, it seems so. 

Your feedback would be appreciated! If we decided to go with a sub-package, I'll cleanup and remove the [module at the root](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4) of the gutenberg directory. I'm not sure how I go about publishing the package later though, but will look into it.

**Comment by fullofcaffeine** [comment]
> Your feedback would be appreciated! If we decided to go with a sub-package, I'll cleanup and remove the [module at the root](https://github.com/WordPress/gutenberg/pull/31732/files#diff-c4339ef01dcf480dbd6ae4307e383b7db7377af08c382f8e48841f6a9b4f52f4) of the gutenberg directory. I'm not sure how I go about publishing the package later though, but will look into it.

I figured this out 💡. It was actually pretty straightforward to reference the internal package without publishing (thanks to Lerna?). I assume we don't want to publish it, so I guess it's fine to keep this way. I tested locally and verified it's working as expected. I think this is ready for another round of reviews. cc @jsnajdr @youknowriad @grzim

File: packages/readable-js-assets-webpack-plugin/test/build.js
Comment (fullofcaffeine) [inline packages/readable-js-assets-webpack-plugin/test/build.js:41]: Not sure why this is failing in GH actions. It is passing locally.

**Comment by jsnajdr** [comment]
> My idea was to avoid manually getting the file contents and writing to the filesystem myself. I thought it'd be simpler to just hook into the lifecycle to add some unminimized assets to the list before terser runs (or after), and then let the already setup WebPack pipeline run after that, thus spitting all the files for us.

That would also work: add a second `.js` asset for each `.min.js` asset, and skip minifying them by configuring Terser with `test: /.min.js$/`. But I think that writing the files directly and not adding them as assets is less risky. We avoid them being post-processed by any other webpack optimization plugin that hooks into the "optimize JS assets" hooks. We also avoid including the assets in the chunk manifests. If we were building an app with this webpack config (instead of libraries), the manifests would include instructions to load _both_ these files.

Instead of direct `fs.writeFile`, we could use webpack's filesystem wrappers -- they add some additional caching, in-memory FS implementations for development mode etc.

File: lib/client-assets.php
Comment (jsnajdr) [inline lib/client-assets.php]: style nit: `an_underscore_convention` is used for identifiers in this PHP script. And I would make the conditional part of the name more minimal, like this:
```
$suffix = $condition ? '.js' : '.min.js'
```

File: lib/client-assets.php
Comment (jsnajdr) [reply lib/client-assets.php]: > I'm not really sure if we should change the filename to `index.min.js` here

Are there places where the exact filename matters? For example, the Gutenberg plugin overrides the Gutenberg assets that are present in Core with the newer ones in the plugin. Does the exact filename play any role there? Or does it involve only symbolic ids of the scripts?

File: packages/readable-js-assets-webpack-plugin/package.json
Comment (jsnajdr) [inline packages/readable-js-assets-webpack-plugin/package.json:31]: We'll need to update the plugin for webpack 5. Instead of the `additionalAssets` hook, we'll likely be hooking into something else. The `terser-webpack-plugin` also uses different hooks for v4 and v5.

File: packages/readable-js-assets-webpack-plugin/index.js
Comment (jsnajdr) [inline packages/readable-js-assets-webpack-plugin/index.js:19]: `compiler.outputFileSystem.writeFile` and `.join` could be used here. Unfortunately `.writeFile` doesn't support promises there.

The Jest test could then use an in-memory FS implementation. But rewriting the test is more like a fun project to play with, not necessary to get this merged.

File: lib/client-assets.php
Comment (fullofcaffeine) [reply lib/client-assets.php]: I don't know. I'll have a look, but I'd love to get insights from @grzim, @ockham or @youknowriad about that.

File: packages/readable-js-assets-webpack-plugin/package.json
Comment (fullofcaffeine) [reply packages/readable-js-assets-webpack-plugin/package.json:31]: Hmm, not sure I understood. Should we do this now or only when we migrate to v5? AFAIK we're still using v4.x, right?

File: lib/client-assets.php
Comment (fullofcaffeine) [reply lib/client-assets.php]: Ah yeah! I'm aware of that, but my muscle habit beat me to it this time.

File: lib/client-assets.php
Comment (fullofcaffeine) [inline lib/client-assets.php]: I'm still confused here and I'm not really sure if we need this ternary logic for the suffix here or if we should hardcode this to the unminized `.js` file, since this function seems related to translations. Any hints?

**Comment by fullofcaffeine** [comment]
@jsnajdr @youknowriad @grzim Could you have another look?

File: lib/client-assets.php
Comment (jsnajdr) [reply lib/client-assets.php]: I believe that `index.min.js` should be hardcoded here. This hook runs in the production plugin, when the `.min.js` files are always present, and checks if the Gutenberg plugin has an override script for a particular Core script, like `wp-data`. Then it will load `wp-data` translations from a `gutenberg` plugin translations file instead of Core one.

File: lib/client-assets.php
Comment (jsnajdr) [inline lib/client-assets.php]: The code that constructs the `index.asset.php` path:
```php
$asset_file = substr( $path, 0, -3 ) . '.asset.php';
```
won't work when `SCRIPT_DEBUG` is on. Because we generate only one asset file named `index.min.asset.php` and here `$asset_file` will be `index.asset.php`. But it doesn't exist.

We should decide whether we want to include `.min` in the asset file path. I'd prefer `index.asset.php`, but the other version also works. And then use it consistently.

**Review by jsnajdr — COMMENTED** [review]
I'm finding only two things that should be improved before this PR can be merged:
- fix the `index(.min).asset.php` situation.
- always use `index.min.js` in `gutenberg_override_translation_file`

A great side benefit of this PR will be that `SCRIPT_DEBUG` starts to work for all Gutenberg JS assets. Until now, as far as I can see, it would load the minified scripts. Except vendor ones like React.

File: lib/client-assets.php
Comment (fullofcaffeine) [reply lib/client-assets.php]: Thanks for explaining, that makes sense 👍🏻

File: lib/client-assets.php
Comment (fullofcaffeine) [reply lib/client-assets.php]: Ah, good point! Thanks for catching this one. I think keeping it as `index.asset.php` is fine. I'll make the change.

File: lib/client-assets.php
Comment (fullofcaffeine) [reply lib/client-assets.php]: Fixed in https://github.com/WordPress/gutenberg/pull/31732/commits/8e689a2edf90dc5595a6bfa49c25b72a9af40f2d.

File: lib/client-assets.php
Comment (fullofcaffeine) [reply lib/client-assets.php]: Fixed in https://github.com/WordPress/gutenberg/pull/31732/commits/49cc7e0ce08a252b7eade5dbab2a71cd1dc78d5b.

File: lib/client-assets.php
Comment (fullofcaffeine) [inline lib/client-assets.php:247]: We could hardcode `index.asset.php` here, but this approach required less changes, and I didn't want to change more than I needed in this context, at least not as part of this PR. Let me know what you think :)

File: packages/dependency-extraction-webpack-plugin/lib/index.js
Comment (jsnajdr) [inline packages/dependency-extraction-webpack-plugin/lib/index.js]: This doesn't work because the `buildFilename` starts with period: `./build/a11y/index.min.js`. I fixed this in 6bcdfc2ae4 with a regexp replacement.

File: lib/client-assets.php
Comment (jsnajdr) [reply lib/client-assets.php:247]: There are asset files that are not `index`, e.g., `frontend.asset.php` for blocks. These are not used in this particular loop but who knows, someday they could. Being flexible sounds OK.

**Review by jsnajdr — APPROVED** [review]
This finally looks good code-wise and the build output looks correct, too. Let's see what the tests say about it. If they are green, I guess this is ready to merge.

File: packages/dependency-extraction-webpack-plugin/lib/index.js
Comment (fullofcaffeine) [reply packages/dependency-extraction-webpack-plugin/lib/index.js]: Thanks 👍🏻

**Comment by fullofcaffeine** [comment]
@jsnajdr Thanks! ❤️  Didn't have much time to look into this one today, but at first glance the failed checks are lint errors that _seem_ unrelated to this change. I'll have a look in my morning, if you don't beat me to it.

**Comment by gziolo** [comment]
Can you also include a change that filters out unminified files from the Compressed size workflow so they don't obscure results?

https://github.com/WordPress/gutenberg/pull/31732#issuecomment-839334100

Screen Shot 2021-06-07 at 09 01 59

File: packages/readable-js-assets-webpack-plugin/README.md
Comment (gziolo) [inline packages/readable-js-assets-webpack-plugin/README.md]:
```suggestion
- It assumes your webpack pipeline is already generating a `.min.js` JS asset file for each JS entry-point.

<br/><br/><p align="center"><img src="https://s.w.org/style/images/codeispoetry.png?1" alt="Code is Poetry." /></p>
```

File: packages/readable-js-assets-webpack-plugin/package.json
Comment (gziolo) [inline packages/readable-js-assets-webpack-plugin/package.json]:
```suggestion
	"version": "1.0.0-prerelease",
```

**Review by gziolo — COMMENTED** [review]
See also guidelines for creating a new package:
https://github.com/WordPress/gutenberg/tree/trunk/packages#creating-a-new-package

- `.npmrc` file is missing
- `CHANGELOG.md` is missing

File: packages/readable-js-assets-webpack-plugin/README.md
Comment (jsnajdr) [reply packages/readable-js-assets-webpack-plugin/README.md]: Just noticed that you spell webpack incorrectly 🙂 They use an all-lowercase spelling: webpack.

**Comment by fullofcaffeine** [comment]
@gziolo Thanks for the review 🙇🏻. This is ready for another look!

**Comment by gziolo** [comment]
Is `@wordpress/readable-js-assets-webpack-plugin` something that can be useful outside of this repository? I understand that creating a new package creates nice isolation of functionality but we can always mark it as private and never publish it to npm if we consider it internal.

**Comment by jsnajdr** [comment]
> Is `@wordpress/readable-js-assets-webpack-plugin` something that can be useful outside of this repository?

It should be useful for any WordPress plugin that builds and ships minified JS files. This plugin will output unminified versions on the side. It's my understanding that it's common for WordPress plugins to publish also unminified versions. For translations, for reviews by the plugin review team, for the `SCRIPT_DEBUG` flag to work...

It could be an option for the `wp-scripts build` command.

**Review by gziolo — APPROVED** [review]

**Comment by talldan** [comment]
Proposing that this is reverted in https://github.com/WordPress/gutenberg/pull/32617, as this seems to stop sourcemaps from working when using `npm run dev`. Unless a very quick fix can be made, revert seems the best option as this will be a frustrating problem for contributors.

**Comment by fullofcaffeine** [comment]
> Proposing that this is reverted in #32617, as this seems to stop sourcemaps from working when using npm run dev. Unless a very quick fix can be made, revert seems the best option as this will be a frustrating problem for contributors.

Quick fix was made [here](https://github.com/WordPress/gutenberg/pull/32621). We'll eventually follow-up with [improvements](https://github.com/WordPress/gutenberg/pull/32621#issuecomment-860824339).

**Pull request merged**
